libfuse/mounter.go

```diff
@@ -98,6 +116,14 @@ func (m *mounter) Unmount() (err error) {
 	return
 }
 
+func (m *mounter) DeleteMountdir() (err error) {
+	err = os.RemoveAll(m.options.MountPoint)
```

Comment:
Hmm, the directory above `kbfs` is used by service for other things in the default mountdirs. But you're right, this could leave additional directories if someone chooses a custom mountdir. I'm not sure how you would know which ones to clean up though. Maybe by saving the filenames on MkdirAll (probably need to write some custom traversal code)?

DeleteMountdir should only be called if the mount exited cleanly. Is it possible for that to happen but the mount is not empty afterwards?

Reply:
Thought about it and I think it's a good idea just in case, added a commit. There's a slight TOCTTOU issue? Should we flock it first or is it ok?

Reply:
Yeah I was thinking recording the created directories. We do need to check to make sure they are empty before removing, so not sure if it's worth it.

Re TOCTTOU, by "rename dance", I meant atomically renaming the directory atomically before doing anything, but I guess there's still no way to make it contingent to the empty check. Flock sounds good to me, but I don't know enough about it. I think Strib did something similar (checking if mount target is empty). @strib do you remember where it was?